README.md

```diff
@@ -63,22 +63,124 @@ hello-world:
+If the branch name is omitted, it will fall back to the default branch name, e.g. if 'master' is the default branch, `@:` is equivalent to `@master:`.
+
+You can use the standard [git revision selection syntax](https://git-scm.com/book/en/v2/Git-Tools-Revision-Selection) to further reduce the selected commits on a release line.
+For example, `@master:v1.0.0..` means "select all tagged commits on the master branch after v1.0.0".
+
+Here a few more examples, demonstrating all potential release specifications:
+
+```yaml
+hello-world:
+  releases:
+  # All tagged commits on the repository's default branch:
+  - git+https://github.com/aiidalab/aiidalab-hello-world.git@:
+  # All tagged commits on the repository's develop branch:
+  - git+https://github.com/aiidalab/aiidalab-hello-world.git@develop:
+  # Specifically the commit tagged with `v1.0.0`:
+  - git+https://github.com/aiidalab/aiidalab-hello-world.git@v1.0.0
+  # All tagged commits on the `master` branch from `v0.1.0` (exclusive) onwards:
+  - git+https://github.com/aiidalab/aiidalab-hello-world.git@master:v0.1.0..
+  # All tagged commits on the `master` branch from `v0.1.0` (inclusive) onwards:
+  - git+https://github.com/aiidalab/aiidalab-hello-world.git@master:v0.1.0^..
+  # All tagged commits on the `master` branch from `v0.1.0` (exclusive) until `v1.0.0`:
+  - git+https://github.com/aiidalab/aiidalab-hello-world.git@master:v0.1.0..v1.0.0
+
+  # The following approaches are *not recommended*, but shown here for completeness:
+  # The latest commit on the repository's default branch
+  - git+https://github.com/aiidalab/aiidalab-hello-world.git
+  # The latest commit on the repository's develop branch:
+  - git+https://github.com/aiidalab/aiidalab-hello-world.git@develop
+```
+
+#### Environment specification
+
+The environment specification is typically directly parsed from the app repository, similar to how [repo2docker](https://repo2docker.readthedocs.io) parses repositories to derive instructions to create a docker image.
+This app store uses the repo2env tool to automatically parse the environment specification for a specific app release.
+
+In the current version, the following files are parsed:
+
+ - `setup.cfg` - The Python requirements listed in that file must be installed within the Python kernel executing the app.
+ - `requirements.txt` (only if setup.cfg does not specify requirements) - The Python requirements listed in that file must be installed within the Python kernel executing the app.
+
+The above listed files are parsed either from a hidden sub-directory called `.aiidalab/` or the repository root directory in that order.
+This means for the purpose of the AiiDAlab environment specification, a file `.aiidalab/requirements.txt` would be preferentially parsed over the `./requirements.txt`.
```

Comment:
is there a specific reason to introduce the additional complexity with the `.aiidalab` directory? (i.e. could one also just live without it?)